x-pack/platform/plugins/shared/fleet/common/types/rest_spec/enrollment_api_key.ts

```diff
@@ -10,7 +10,9 @@ import type { EnrollmentAPIKey } from '../models';
 import type { ListResult, ListWithKuery } from './common';
 
 export interface GetEnrollmentAPIKeysRequest {
-  query: ListWithKuery;
+  query: ListWithKuery & {
+    showHidden?: boolean;
```

Comment:
Not sure what is the best way to name this one, we are not super consistent with those flags, we have a bunch of show* some hide*

The default behaviour here is to show the hidden enrolment keys to avoid a breaking change

Reply:
we don't necessarily need a new API field, we could just pass the KQL to the `kuery`

Reply:
That's true, though from the experience with `showInactive` in the agents API, it seemed to bring more complexity to maintain than expected.

Reply:
@jen-huang curious to have your thoughts on that one has you probably had the same concerns to hide agentless agent/agent policy

Reply:
@nchaulet for hiding agent policies, I was just going to pass it in the KQL since the property exists. I haven't looked yet at if agents already have some metadata property that marks it as an agentless agent, if so I will also use KQL, otherwise I will add a param if the lookup logic is more complicated

for enrollment keys I thought a param would work without introducing a new `hidden` field.  if we decide that's the route to go, I'm in favor of just using KQL for the filtering

btw, I don't see anything in this PR that backfills `hidden` field for existing enrollment keys, should we consider doing that?

Reply:
Perfect I will do the change to use a kuery

> btw, I don't see anything in this PR that backfills hidden field for existing enrollment keys, should we consider doing that?

It's not a saved object but a fleet system indices so it's not easy to do a backfill, moreover, I added some code to do the client side filtering as we do for managed policy today for old enrollment api keys.